### PR TITLE
Border Style Control: Update styling for consistency with border width control

### PR DESCRIPTION
**Pull request by andrewserong**
## Description

Raised in https://github.com/WordPress/gutenberg/pull/34061#issuecomment-989499116 by @ramonjd and following on from the PR by @aaronrobertshaw to standardise ToolsPanel input margins (https://github.com/WordPress/gutenberg/pull/36387), this PR looks at updating the Border Style Control to have consistent margin, padding and line-height as the border width control.

## How has this been tested?

1. Running a blocks-based theme that opts-in to border support (e.g. Twenty Twenty Two) add a group block
2. Enable the border style in the Border tools panel in the block inspector
3. Before this PR, note that the Style buttons and legend aren't quite aligned with the border width control
4. With this PR applied, they should be aligned

## Screenshots 

| Before | After |
| --- | --- |
| image | image |

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 ### Bug Fix
 
+-   Fix alignment of `BorderStyleControl` legend with border width control's label ([#37244](https://github.com/WordPress/gutenberg/pull/37244)).
```
Comment (ciampo) [inline packages/components/CHANGELOG.md]: The `BorderStyleControl` component doesn't belong to the `@wordpress/components` package, and therefore we should not add a CHANGELOG entry there for this PR.

**Review by ciampo — APPROVED** [review]
Thank you for the ping, Andrew!

Those changes do fix the alignment issue! I believe that this PR is good to go (after the CHANGELOG entry is removed, as explained in the comment below).

---

I also have some feedback/question on the `BorderStyleControl` component — again, definitely something that can/should be tackled separately:

- As also highlighted in https://github.com/WordPress/gutenberg/pull/36387#pullrequestreview-807268145, the proper fix in my opinion is that all components in `@wordpress/components` get refactored to be based off the same underlying `BaseControl`, which would also imply that they share the same implementation for their label. If that was the case, we would then be able to take that base control and use it also in custom controls (like in this example), and get a perfect label alignment out of the box.
- Have you considered using [`ToggleGroupControl`](https://wordpress.github.io/gutenberg/?path=/story/components-togglegroupcontrol--default) instead of using `Button`s ?
- The component is using some classnames (e.g. `components-border-style-control`) that use the same naming convention used by the legacy components in `@wordpress/components`, and therefore somehow imply that this component is part of that package. As we migrate away from hardcoded `component-*` classnames in `@wordpress/components`, the fact that the same naming convention is being used in another package will probably add extra confusion. I also know that this comment would apply to most components defined in `block-editor/src/components`.

**Comment by aaronrobertshaw** [comment]
I haven't had a chance to review this yet but can answer one of the questions above now:

> Have you considered using ToggleGroupControl instead of using Buttons ?

The control needs to be able to toggle off a selection. The `ToggleGroupControl` uses a `RadioGroup` so I don't believe we can use it. I think this question came up recently in regards to the letter case and text-decoration controls within the typography panel as well.

**Review by ramonjd — APPROVED** [review]
Thanks for looking at this so quickly!

**Before**

Screen Shot 2021-12-10 at 9 07 35 am

**After**
Screen Shot 2021-12-10 at 9 08 22 am

File: packages/components/CHANGELOG.md (hunk 1 above)
Comment (andrewserong) [reply packages/components/CHANGELOG.md]: Thanks! Removed.

**Comment by andrewserong** [comment]
> The BorderStyleControl component doesn't belong to the @wordpress/components package, and therefore we should not add a CHANGELOG entry there for this PR.

Thanks for the review @ciampo! And apologies, I was working too quickly here and saw `components` in the path and forgot to double-check which package I was in 😅

> the proper fix in my opinion is that all components in `@wordpress/components` get refactored to be based off the same underlying BaseControl, which would also imply that they share the same implementation for their label. 

Agreed! Once we've stabilised some of these individual cases for 5.9 it'd be good for us to do this refactor to tidy up the components across the board 👍

**Pull request merged**